Lib/test/test_email/test_headerregistry.py

```diff
@@ -1648,6 +1648,13 @@ def test_fold_overlong_words_using_RFC2047(self):
                 'xxxxxxxxxxxxxxxxxxxx=3D=3D-xxx-xx-xx?=\n'
             ' =?utf-8?q?=3E?=\n')
 
+    def test_message_id_header_is_not_folded(self):
+        h = self.make_header(
+            'Message-ID',
+            '<somemessageidlongerthan@maxlinelength.com>')
+        self.assertEqual(
+            h.fold(policy=policy.default.clone(max_line_length=20)),
+            'Message-ID: <somemessageidlongerthan@maxlinelength.com>\n')
```

Comment:
Should have another test with a no-fold-literal in it.  And what happens if we get a header with non-ascii in the message-id?

Reply:
Thanks for pointing it out!

As it turns out, the message-id not folding was completely by chance since all the tested subtokens weren't allowed to be folded and hence the header wasn't folding. As soon as I added a test with non-ascii characters, they started to fold.

I read through `_refold_parse_tree` to figure out if there is a good way to prevent folding of **subparts** when the part itself isn't allowed to be folded. See my new change for what I ended up doing. 

A side effect of which is that the Header does get wrapped if it exceeds the max_ilne_length but the msg-id token is kept intact (which will be needed when parsing `References` header). We also don't do any decoding of non-ascii parts. 

I did another implementation where I added `fold` method to `Message-ID` header to just return `str(header) + policy.linesep`, but that seemed wrong since there was possibility that we could fold at the boundary between `Message-ID: ` and `value`. Also, that probably won't work well for other Header types like `References` which would have multiple `msg-id` tokens.

What do you think about the current implementation? It feels a bit hacky to be honest, but I can't find a better way.

Reply:
I am slightly confused with the actual RFC text, since it says:

   > Unlike addr-spec, this syntax only permits the dot-atom-text
   > form on the left-hand side of the "@" and does not have internal CFWS
   > anywhere in the message identifier.

But it also allows `obs-id-left`  and `obs-id-right` which is allowed to be folded using encoded words.  So do we allow folding of subparts or not?

Also not sure if leaving non-ascii text as is great idea.

Reply:
I think what you want is a fold method on MsgID that returns str(self) (with a comment that msg-ids may not be folded).  I unfortunately can't remember what the Right Thing is to do with non-ascii, so I'll probably have to take a deeper look at that.